tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,147 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput } from '../components';
+
+type modularShieldGeneratorData = {
```

Comment:
in local files, I think short names are fine. Data is totally acceptable.
the other issue with this is that (by convention, no real rule,) types are PascalCase.